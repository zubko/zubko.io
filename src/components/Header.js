import React from 'react';
import PropTypes from 'prop-types';

import { Link, Menu } from '../components';
import { Colors } from '../Theme';

const Header = ({ siteTitle }) => (
  <div css={styles.headerOutter}>
    <div css={styles.headerInner}>
      <Link to="/" css={styles.home}>
        {siteTitle}
      </Link>
      <Menu />
    </div>
  </div>
);

const styles = {
  headerOutter: {
    background: Colors.main,
    marginBottom: `1.45rem`,
  },
  headerInner: {
    margin: `0 auto`,
    maxWidth: 960,
    padding: `1.45rem 1.0875rem`,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  home: {
    color: `white`,
    textDecoration: `none`,
    ':hover': {
      color: `white`,
      textDecoration: `none`,
    },
    fontSize: '1.5rem',
    fontWeight: 'bold',
  },
};

Header.propTypes = {
  siteTitle: PropTypes.string,
};

Header.defaultProps = {
  siteTitle: ``,
};

export default Header;
