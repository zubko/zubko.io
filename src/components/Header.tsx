/**
 * Top common part of the website
 */

import { graphql, StaticQuery } from "gatsby";
import { useState } from "react";

import { Link, MenuHorizontal, MenuVertical } from "../components";
import { Colors, Media } from "../Theme";
import { rhythm } from "../Typography";

const MENU_MEDIA_SPLIT = 780;
const MediaForVertMenu = Media.lessThanPx(MENU_MEDIA_SPLIT);
const MediaForHorzMenu = Media.greaterThanPx(MENU_MEDIA_SPLIT);

type Props = {
  siteTitle?: string;
};

const Header = ({ siteTitle = "" }: Props) => {
  const [isMenuOpened, setMenuOpened] = useState(false);
  return (
    <StaticQuery<Queries.MenuQueryQuery>
      query={graphql`
        query MenuQuery {
          allMenuYaml {
            edges {
              node {
                path
                title
              }
            }
          }
        }
      `}
      render={({ allMenuYaml: { edges: menuEdges } }) => {
        const menuLinks = menuEdges.map(e => e.node);
        return (
          <div css={styles.headerOuter}>
            <div css={styles.headerInner}>
              <Link usePlainStyle to="/" css={styles.homeLink}>
                <h1>{siteTitle}</h1>
              </Link>

              <MenuHorizontal
                css={{ [MediaForVertMenu]: { display: "none" } }}
                links={menuLinks}
              />
              <MenuVertical
                isHead
                isOpened={isMenuOpened}
                setOpened={setMenuOpened}
                css={{ [MediaForHorzMenu]: { display: "none" } }}
                links={menuLinks}
              />
            </div>
            <MenuVertical
              isBody
              isOpened={isMenuOpened}
              setOpened={setMenuOpened}
              css={{ [MediaForHorzMenu]: { display: "none" } }}
              links={menuLinks}
            />
          </div>
        );
      }}
    />
  );
};

const styles = {
  headerOuter: {
    marginBottom: `${rhythm(0.5)}`
  },
  headerInner: {
    paddingTop: rhythm(0.5),
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  homeLink: {
    textDecoration: `none`,
    ":hover": {
      textDecoration: `none`
    },
    background: "none",
    textShadow: "none",
    "> h1": {
      color: Colors.mainDarker,
      margin: 0,
      fontWeight: 600,
      fontSize: "1.21225rem" // h3
    }
  }
};

export default Header;
