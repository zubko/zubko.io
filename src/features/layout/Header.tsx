/**
 * Top common part of the website
 */

import { graphql, StaticQuery } from "gatsby";
import { useState } from "react";
import styled from "styled-components";

import { Link } from "../../components/Link";
import { Colors, Media } from "../../Theme";
import { rhythm } from "../../Typography";

import { MenuHorizontal as MenuHorizontalOrig } from "./MenuHorizontal";
import { MenuVertical as MenuVerticalOrig } from "./MenuVertical";

const MENU_MEDIA_SPLIT = 780;
const MediaForVertMenu = Media.lessThanPx(MENU_MEDIA_SPLIT);
const MediaForHorzMenu = Media.greaterThanPx(MENU_MEDIA_SPLIT);

type Props = {
  siteTitle?: string;
};

type LinkNode = {
  path: string;
  title: string;
};

export const Header = ({ siteTitle = "" }: Props) => {
  const [isMenuOpened, setMenuOpened] = useState(false);
  return (
    <StaticQuery<Queries.HeaderMenuQuery>
      query={graphql`
        query HeaderMenu {
          allMenuYaml {
            edges {
              node {
                path
                title
              }
            }
          }
        }
      `}
      render={({ allMenuYaml: { edges: menuEdges } }) => {
        const menuLinks = menuEdges.map((e) => e.node as LinkNode);
        return (
          <HeaderOuter>
            <HeaderInner>
              <HomeLink usePlainStyle to="/">
                <h1>{siteTitle}</h1>
              </HomeLink>

              <MenuHorizontal links={menuLinks} />
              <MenuVertical
                isHead
                isOpened={isMenuOpened}
                onOpened={setMenuOpened}
                links={menuLinks}
              />
            </HeaderInner>
            <MenuVertical
              isBody
              isOpened={isMenuOpened}
              onOpened={setMenuOpened}
              links={menuLinks}
            />
          </HeaderOuter>
        );
      }}
    />
  );
};

const HeaderOuter = styled.div`
  margin-bottom: ${rhythm(0.5)};
`;

const HeaderInner = styled.div`
  padding-top: ${rhythm(0.5)};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const HomeLink = styled(Link)`
  text-decoration: none;
  :hover {
    text-decoration: none;
  }
  background: none;
  text-shadow: none;
  > h1 {
    color: ${Colors.mainDarker};
    margin: 0;
    font-weight: 600;
    font-size: 1.21225rem; // h3
  }
`;

const MenuHorizontal = styled(MenuHorizontalOrig)`
  ${MediaForVertMenu} {
    display: none;
  }
`;

const MenuVertical = styled(MenuVerticalOrig)`
  ${MediaForHorzMenu} {
    display: none;
  }
`;
