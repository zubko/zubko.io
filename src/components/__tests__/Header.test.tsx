import React from "react";
import renderer from "react-test-renderer";
import { StaticQuery } from "gatsby";

import Header from "../Header";

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      allMenuYaml: {
        edges: [
          {
            node: {
              path: "/path-1/",
              title: "title-1",
            },
          },
          {
            node: {
              path: "/path-2/",
              title: "title-2",
            },
          },
        ],
      },
    })
  );
});

describe("Header", () => {
  it("renders without regressions", () => {
    const tree = renderer.create(<Header siteTitle="siteTitle" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
