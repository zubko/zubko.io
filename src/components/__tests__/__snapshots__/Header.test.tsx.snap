// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without regressions 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1pcmc93{margin-bottom:0.809rem;}",
      }
    }
    data-emotion="css 1pcmc93"
  />,
  <div
    className="css-1pcmc93"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-13o4jer{padding-top:0.809rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        }
      }
      data-emotion="css 13o4jer"
    />
    <div
      className="css-13o4jer"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-nauaj9{-webkit-text-decoration:none;text-decoration:none;background:none;text-shadow:none;}.css-nauaj9:hover{-webkit-text-decoration:none;text-decoration:none;}.css-nauaj9 >h1{color:#2C8127;margin:0;font-weight:600;font-size:1.21225rem;}",
          }
        }
        data-emotion="css nauaj9"
      />
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-4afkbd{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;background:none;text-shadow:none;}.css-4afkbd:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}.css-4afkbd:hover{-webkit-text-decoration:none;text-decoration:none;}.css-4afkbd >h1{color:#2C8127;margin:0;font-weight:600;font-size:1.21225rem;}",
          }
        }
        data-emotion="css 4afkbd"
      />
      <a
        className="css-4afkbd"
        href="/"
      >
        <h1>
          siteTitle
        </h1>
      </a>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "@media (max-width: 779px){.css-1vi9qj2{display:none;}}",
          }
        }
        data-emotion="css 1vi9qj2"
      />
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1nomk7v-MenuHorizontal{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}@media (max-width: 779px){.css-1nomk7v-MenuHorizontal{display:none;}}",
          }
        }
        data-emotion="css 1nomk7v-MenuHorizontal"
      />
      <div
        className="css-1nomk7v-MenuHorizontal"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-1mgyam6{-webkit-text-decoration:none;text-decoration:none;font-size:1rem;font-weight:medium;background:none;}",
            }
          }
          data-emotion="css 1mgyam6"
        />
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-t9hu3x{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;font-size:1rem;font-weight:medium;background:none;}.css-t9hu3x:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}",
            }
          }
          data-emotion="css t9hu3x"
        />
        <a
          className="css-t9hu3x"
          href="/path-1/"
        >
          title-1
        </a>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-1u0gifa-Separator{color:white;margin-left:0.5rem;margin-right:0.5rem;cursor:default;}",
            }
          }
          data-emotion="css 1u0gifa-Separator"
        />
        <span
          className="css-1u0gifa-Separator"
        >
           
        </span>
        <a
          className="css-t9hu3x"
          href="/path-2/"
        >
          title-2
        </a>
      </div>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "@media (min-width: 780px){.css-1dhntwi{display:none;}}",
          }
        }
        data-emotion="css 1dhntwi"
      />
      <div
        className="css-1dhntwi"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-50n0xx-MenuVertical{cursor:pointer;color:#2C8127;border:1px solid #2C8127;padding:0.5rem;background-color:transparent;}.css-50n0xx-MenuVertical:focus{outline:0;}",
            }
          }
          data-emotion="css 50n0xx-MenuVertical"
        />
        <button
          className="css-50n0xx-MenuVertical"
          onClick={[Function]}
        >
          Menu
        </button>
      </div>
    </div>
    <div
      className="css-1dhntwi"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1tobc0i-MenuVertical{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-bottom:0.809rem;padding-top:1.618rem;display:none;}",
          }
        }
        data-emotion="css 1tobc0i-MenuVertical"
      />
      <div
        className="css-1tobc0i-MenuVertical"
      >
        <a
          className="css-t9hu3x"
          href="/path-1/"
        >
          title-1
        </a>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-l1p1i0-Separator{height:0.5rem;color:white;cursor:default;}",
            }
          }
          data-emotion="css l1p1i0-Separator"
        />
        <span
          className="css-l1p1i0-Separator"
        >
           
        </span>
        <a
          className="css-t9hu3x"
          href="/path-2/"
        >
          title-2
        </a>
      </div>
    </div>
  </div>,
]
`;
